tests/buildenv.sh

```diff
@@ -2,12 +2,67 @@
 
 set -eu
 
-PACKAGES="apk mksquashfs qemu-arm xz"
+ARM_EMU_BIN="${ARM_EMU_BIN:-}"
+BINFMT_MISC="${BINFMT_MISC:-/proc/sys/fs/binfmt_misc/}"
+PACKAGES="apk mksquashfs xz"
 FILESYSTEMS="overlay squashfs tmpfs"
 
+ARMv7_MAGIC="7f454c4601010100000000000000000002002800"
+
 result=0
 
 
+check_emulation_support()
```

Comment:
This function is too long and is difficult to understand, could split it up into several self explaining helper functions to improve readability.